automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala

```diff
@@ -63,6 +63,7 @@ class WorkspaceSummaryPage(namespace: String, name: String)(implicit webDriver:
 
     def clickEdit(): Unit = {
       editButton.doClick()
+      saveButton.awaitEnabledState()
```

Comment:
Ah, nice find!

So, this is an interesting page in that it switches between view/edit modes. The `awaitReady` function is not aware of these modes so it cannot behave appropriately.  I agree that waiting for the save button is a good solution.

My only question is whether it should be here or if `clickEdit` should manipulate a property of the page that `awaitReady` can use to decide what it's going to check. Advantage is that it cohesively describes the stateful nature of the page. Disadvantage is that is an extra level of indirection when tracing the linear behavior of a scenario.

I can't argue strongly for either case over the other at the moment. Curious to hear what others think. For now, either way is fine with me.

Reply:
I'm working on a better way to do this whole page. I think this is the most correct option with the way the code is right now.